test/time_index_conversions.jl

```diff
@@ -37,6 +37,40 @@ end
     end
 end
 
+@testset "RoundFullyContainedSampleSpans" begin
+    input = TimeSpan(0, Second(30))
+    aligned = AlignedSpan(1 / 30, input, RoundFullyContainedSampleSpans)
+    @test aligned == AlignedSpan(1 / 30, 1, 1)
+
+    input = TimeSpan(0, Second(30) + Nanosecond(1))
+    aligned = AlignedSpan(1 / 30, input, RoundFullyContainedSampleSpans)
+    @test aligned == AlignedSpan(1 / 30, 1, 1)
+
+    input = TimeSpan(0, Second(59))
+    aligned = AlignedSpan(1 / 30, input, RoundFullyContainedSampleSpans)
+    @test aligned == AlignedSpan(1 / 30, 1, 1)
+
+    input = TimeSpan(0, Second(60))
+    aligned = AlignedSpan(1 / 30, input, RoundFullyContainedSampleSpans)
+    @test aligned == AlignedSpan(1 / 30, 1, 2)
+
+    input = TimeSpan(Nanosecond(1), Second(60) + Nanosecond(1))
+    aligned = AlignedSpan(1 / 30, input, RoundFullyContainedSampleSpans)
+    @test aligned == AlignedSpan(1 / 30, 2, 2)
+
+    input = TimeSpan(0, Second(60) + Nanosecond(1))
+    aligned = AlignedSpan(1 / 30, input, RoundFullyContainedSampleSpans)
+    @test aligned == AlignedSpan(1 / 30, 1, 2)
+
+    input = TimeSpan(0, Second(60))
+    aligned = AlignedSpan(1 / 30, input, RoundFullyContainedSampleSpans)
+    @test aligned == AlignedSpan(1 / 30, 1, 2)
+
+    # No spans contained
+    input = TimeSpan(Nanosecond(1), Second(30) + Nanosecond(1))
+    @test_throws ArgumentError AlignedSpan(1 / 30, input, RoundFullyContainedSampleSpans)
```

Comment:
do we really want this to be an error?  or should it be like an empty span somehow?  do we throw on any other failures to construct an appropriate span?

Reply:
I pushed a commit to just use the check in the inner constructor, which has a more informative error message anyway. This way also the PR diff makes it clear the error is not new.